skip process isolation check for transmit and process

TheRealHaoLiu opened the pull request:
https://github.com/ansible/ansible-runner/issues/1085

TheRealHaoLiu committed:
skip process isolation check for transmit and process

moving down the process isolation executable so that transmit and
process no longer checks for its existance

ansible-runner worker will still run the check since it will invoke
ansible-runner run with process-isolation-executable and it will still
be checked properly

Co-Authored-By: Jeff Bradberry <685957+jbradberry@users.noreply.github.com>
Signed-off-by: Hao Liu <haoli@redhat.com>

shanemcd commented on .gitignore:
```diff
@@ -28,6 +28,7 @@ rpm-build/
 .tox/
 .venv
 venv/
+.vscode/
```
Can you split this out into a different commit?

Shrews replied:
This should be in the user's `$HOME/.gitignore_global` file, IMO, not a per-repo ignore.

matburt commented on ansible_runner/interface.py:
```diff
@@ -83,7 +83,7 @@ def init_runner(**kwargs):
             output.set_logfile(logfile)
 
-    if kwargs.get("process_isolation", False):
+    if kwargs.get("process_isolation", False) and kwargs.get("streamer") not in ('transmit', 'process'):
```
I have this in my PR also: https://github.com/ansible/ansible-runner/pull/1079/files#diff-179ccecedba1c6ccf73b8c02f0321b252c018226e41ce82887a93ace762d2fbbR55

This line causes an astonishing amount of problems when interfacing with Runner from the CLI

matburt replied:
I think the reason it does this has to do with the rest of my PR, which is that the CLI entrypoint clashes with the module level entrypoint when managing process isolation paths.

My work on #1079  focuses on realigning those two paths.

matburt commented:
Fix that vscode comment and then I think we should merge this asap.

TheRealHaoLiu commented:
`.gitignore` change amended as per suggestion

TheRealHaoLiu commented:
code change after peer programing with @jbradberry moved the location of the check

Shrews commented:
Manually tested this to verify that `worker` will bail if the executable is not found. I'd like to have a test for this before merging it. Thanks!

TheRealHaoLiu committed:
add test_process_isolation_executable_not_exist

test_process_isolation_executable_not_exist will validate that
- transmit will run when process isolation executable missing and properly pass on the related kwargs to the output
- worker will fail with sys.exit(1) when process isolation executable missiing

Signed-off-by: Hao Liu <haoli@redhat.com>

TheRealHaoLiu commented:
> Manually tested this to verify that `worker` will bail if the executable is not found. I'd like to have a test for this before merging it. Thanks!

testcase added

Shrews commented on test/integration/test_transmit_worker_process.py:
```diff
@@ -168,6 +168,54 @@ def process_method(results_sockfile_read):
+        assert transmitter.kwargs['process_isolation_executable'] == job_kwargs['process_isolation_executable']
+
+        # validate that transmit did not fail due to missing process isolation executable
+        assert transmitter.rc in (None, 0)
```
Is `None` really a valid value here?

TheRealHaoLiu replied:
https://github.com/ansible/ansible-runner/blob/devel/test/integration/test_transmit_worker_process.py#L74-L75

Shrews commented on test/integration/test_transmit_worker_process.py:
```diff
@@ -168,6 +168,54 @@ def process_method(results_sockfile_read):
+                assert kwargs['process_isolation'] is True
+                assert kwargs['process_isolation_executable'] == process_isolation_executable
+                break
+
```
Similar to the `transmit` check above, you should check that `worker` will fail if the executable is missing.

Shrews commented on test/integration/test_transmit_worker_process.py:
```diff
@@ -168,6 +168,54 @@ def process_method(results_sockfile_read):
+        """
+        def mock_check_isolation_executable_installed(executable):
+            if executable == "exist":
+                return True
```
```suggestion
                return True
            return False
```

TheRealHaoLiu commented:
testcase commit amended base on @Shrews's feedback

Shrews reviewed (approved):
Thanks for the PR! If you want this in any of the previously released 2.x versions, you'll need to supply a backport to the appropriate release branches. We follow the same basic backport process as Ansible (https://docs.ansible.com/ansible/latest/community/development_process.html#backport-process).

TheRealHaoLiu commented:
> Thanks for the PR! If you want this in any of the previously released 2.x versions, you'll need to supply a backport to the appropriate release branches. We follow the same basic backport process as Ansible (https://docs.ansible.com/ansible/latest/community/development_process.html#backport-process).

no need for backport thanks

TheRealHaoLiu commented:
backport to 2.2 https://github.com/ansible/ansible-runner/pull/1094
